tests/foreman/ui/test_role.py

```diff
@@ -99,14 +102,127 @@ def test_positive_update_permission(self):
         @Assert: Role is updated
         """
         name = gen_string('alpha')
+        resource_type = 'Architecture'
+        permissions = ['view_architectures', 'edit_architectures']
         with Session(self.browser) as session:
             make_role(session, name=name)
             self.assertIsNotNone(self.role.search(name))
             self.role.add_permission(
                 name,
-                resource_type='Architecture',
-                permission_list=['view_architectures', 'create_architectures'],
+                resource_type=resource_type,
+                permission_list=permissions,
+            )
+            self.assertIsNotNone(
+                self.role.wait_until_element(common_locators['alert.success']))
+            new_permissions = self.role.get_permissions(name, resource_type)
+            self.assertIsNotNone(new_permissions)
+            self.assertEqual(set(permissions), set(new_permissions))
+
+    @tier1
+    def test_positive_clone_builtin(self):
+        """Clone one of the builtin roles
+
+        @id: e3e6af90-fb31-4de9-8f36-f50550d7f00e
+
+        @Assert: New role is created.
+        """
+        builtin_name = choice(ROLES)
+        new_name = gen_string('alpha')
+        with Session(self.browser):
+            self.role.clone(builtin_name, new_name)
+            self.assertIsNotNone(
+                self.role.wait_until_element(common_locators['alert.success']))
+            self.assertIsNotNone(self.role.search(new_name))
+            # Ensure that cloned role contains correct resource type
```

Comment:
In comment, we have `resource type` in singular form, but then `builtin_resources` and `cloned_resources` in plural. A little bit confusing fact